machine.go

```diff
@@ -641,6 +661,41 @@ func (machine *Machine) toProto() *v1.Machine {
 	return machineProto
 }
 
+// getTags will return the tags of the current machine.
+func getTags(
+	aclPolicy ACLPolicy,
+	machine Machine,
+	stripEmailDomain bool,
+) (validTags []string, invalidTags []string) {
+	validTagMap := make(map[string]bool)
+	invalidTagMap := make(map[string]bool)
+	for _, tag := range machine.HostInfo.RequestTags {
+		owners, err := expandTagOwners(aclPolicy, tag, stripEmailDomain)
```

Comment:
This whole part of the invalidTagMap really confuses me, and I cant really test what happens from the tests, could you try to elaborate?

Reply:
What is bothering you ? The fact that I use a map or the concept of invalidTags ? I'm using a map to remove duplicated tags at insertion. It is possible for a machine to have multiple times the same tag (declared in `tailscale up`).

Reply:
Right, so an invalid tag is a tag added by a user that does'nt have permission to add tags. If joe (a simple user) can setup his computer to be a server, then he can talk to the database directly even if he should not. So the validation of the tags are made with the TagOwners in the ACL file. The documentation of this feature has not been made yet. It seems necessary.

Reply:
I feel that we need to explain this somewhere. Do you have any good location for this ?

Reply:
hmm, maybe in the signature of the function 🤔